### PR TITLE
Fix bugged winter tiles Fixes #494

### LINKED ISSUE #494: fix winter landscape
### Expected results

I can reach winter landscape tile 1 , left to the goal tile

### Actual results

Unreachable and broken svg
https://github.com/kubami9/labyrinth-km-contribute/blob/cc3fe220100eafd6edeb91c2fdd8c80b6c51ebd0/tiles/rooms/floor/winter1.svg

### Steps to reproduce the issue

right up right up up - then, left is not working

### Implementation/Solution Ideas

- add door on other side of this landscape tile
- fix svg

Related: #466

### PR DESCRIPTION
**Pull request by kubami9**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a preview link and screenshot showing after and before the changes.
- [x] Included a description of change below.
- [ ] Squashed the commits.

### Changes done in this Pull Request

- If your change will be reflected on the website, please provide a **Test-Link**
  https://kubami9.github.io/labyrinth-km-contribute/

  Fixes #494

### The problem I want to solve or the facility I want to improve is/are
Fixed winter landscape. Enabled moving in proper directions and changed `winter2` tile place. Also fixed broken SVG. I'm going to add animation to it in the next PR.

### My steps to solve it
Enabled moving in right direction from minecraft tile and enabled proper directions in winter tiles. Deleted code braking svg file which I previously added by mistake.

### Screenshots, (winter tiles marked with the red circles)
pr_screen

### CODE REVIEW

**Comment by niccokunzmann** [comment]
@kubami9 Thanks, this does not fix the issue.

From description:
Issues to fix:
- can not leave & enter on top-right from snow tile

Additional issue:
- can move though wall from minecraft tile

**Comment by niccokunzmann** [comment]
I will check the cod enow.

File: js/tiles.js [hunk 1]
```diff
@@ -781,10 +781,10 @@ const door = {
     }),
     minecraft: Object.assign({}, OpenDoors, {
         canEnterFromTheRight() {
-            return false;
+            return true;
```
Comment (niccokunzmann) [inline js/tiles.js]: this should be false as there is a wall

File: js/tiles.js [hunk 2]
```diff
@@ -781,10 +781,10 @@ const door = {
...
         },
         canLeaveToTheRight() {
-            return false;
+            return true;
```
Comment (niccokunzmann) [inline js/tiles.js]: this should be false as there is a wall

File: js/tiles.js [hunk 3]
```diff
@@ -1005,13 +1005,25 @@ const door = {
...
             return false;
         },
+        canEnterFromTheLeft: function(player) {
+            return true;
+        },
         canLeaveToTheRight: function(player) {
             return false;
         },
+        canLeaveToTheBottom: function(player) {
+            return false;
```
Comment (niccokunzmann) [inline js/tiles.js]: this should be true as there is no wall in the tile. It is open there.

File: js/tiles.js [hunk 4]
```diff
@@ -1005,13 +1005,25 @@ const door = {
...
+        },
+        canLeaveToTheLeft: function(player) {
+            return true;
```
Comment (niccokunzmann) [inline js/tiles.js]: that is correct!

File: js/levels.js [hunk 5]
```diff
@@ -11,8 +11,8 @@
 function createFirstLevel()
 {
   return new Level("Starter", [
-    [door.none, door.minecraftEntry, door.minecraft, door.winter1, door.goal, door.winter2],
-    [door.none, door.orchestra, door.Forbidden, door.chocolate, door.guitarCase, door.none],
+    [door.none, door.minecraftEntry, door.minecraft, door.winter1, door.goal, door.none],
```
Comment (niccokunzmann) [inline js/levels.js]: You can add an additional line of door.none to the top.

**Comment by niccokunzmann** [comment]
For understanding: When the player wants to move somewhere, the it asks the tile it wants to leave if it can leave in this direction and the tile it wants to enter if it can enter from this direction.

https://github.com/fossasia/labyrinth/blob/master/js/player.js#L41

All tiles are open on the bottom and left. Thus, there they can always be entered.

**Comment by kubami9** [comment]
Would it be okay? Left winter tile lets you go only where the doors are and the right one to top left and bottom left. Minecraft section would be left as it is now in the game since it has no doors.
testpr
Ping: @niccokunzmann

**Comment by niccokunzmann** [comment]
> Left winter tile lets you go only where the doors are

If I understand it like this: Left winter tile is open from top, left and bottom, but closed from right. This is what I see so I think it is ok.

> right one to top left and bottom left.

On the right winter tile, you can go to bottom and left. Sounds good.

> Minecraft section would be left as it is now in the game since it has no doors.

I agree

---

Also, in the picture, I see that maybe the way to the goal tile is blocked. Please allow reaching this garden goal tile.

Does it sound like I understood?

**Comment by kubami9** [comment]
As you wrote but left tile is open only from top and bottom since it has door only in these directions.

And right tile is open on the left and top so `goal` tile is reachable.

testpr

**Comment by niccokunzmann** [comment]
Thanks for painting that. This seems to be what I said except one thing:

shouldn't the right winter tile also allow access to the right where no tile is shown right now?

If yes, we 100% agree

**Comment by kubami9** [comment]
But there's no tile, it was door.none until I changed it for door.winter2

no_tile

**Comment by kubami9** [comment]
Is it ready to merge now? @niccokunzmann

**Comment by niccokunzmann** [comment]
As you say. I do not want to be in the way of merging so I ask. I think, you checked it and played it because I will not check every time. Thanks, please check and open issues if something in the labyrinth needs improvement in your eyes.

**Comment by niccokunzmann** [comment]
I will check travis

**Comment by niccokunzmann** [comment]
great, all fine!

**Comment by kubami9** [comment]
Thank you 😄 I'll be fixing if I realise any bug

**Pull request merged**